src/main/bx/bifs/aiDocuments.bx

```diff
@@ -0,0 +1,59 @@
+/**
+ * [BoxLang]
+ *
+ * Copyright [2023] [Ortus Solutions, Corp]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
```

Comment:
This is the main entry point now for loaders.  You pass in the source and then a config. It no longer loads immediately.  It's a fluent document loader builder.

// Load documents
docs = aiDocument( "/docs", { type: "markdown", chunkSize: 500 } ).load();

// Advanced fluent config
docs = aiDocument( "/docs", { type: "markdown" } )
    .recursive()
    .chunkSize( 500 )
    .load();

// One-shot to memory
result = aiDocument( "/docs", { type: "markdown", chunkSize: 500 } )
    .toMemory( myMemory );

This means that the DocumentLoader interface needs consistency on methods like 
- load()
- loadAsStream()
- loadAsync()

And config options.